internal/exec/stages/files/passwd.go

```diff
@@ -41,11 +41,11 @@ func (s *stage) expandGlobList(globs ...string) ([]string, error) {
 
 // createPasswd creates the users and groups as described in config.Passwd.
 func (s *stage) createPasswd(config types.Config) error {
-	if err := s.createGroups(config); err != nil {
+	if err := s.ensureGroups(config); err != nil {
 		return fmt.Errorf("failed to create groups: %v", err)
```

Comment:
Could change the error message for both users and groups to something like: `failed to configure X: %v`